pkg/encoding/v1/encoding_test.go

```diff
@@ -419,5 +419,44 @@ volumes: []
 			continue
 		}
 	}
+}
```

Comment:
I don't like these unit test. We do table driven testing (a bit of Golang way) everywhere else.

There should be 2 unit tests for:
 - all the test cases (nil, 0, 42, fail, ...) - TestService_UnmarshalYAML (you can get inspiration [here](https://github.com/tnozicka/opencompose/blob/a21430158830bff240a8252aa4b609e994907238/pkg/encoding/v1/encoding_test.go#L118))
The unit test for service hasn't been there before which is my fault, but If you wouldn't mind adding it there, that will be great 
 - to ensure the value gets copied (probably just add one case to existing test) -https://github.com/surajssd/opencompose/blob/8e34ab28588e02b85fada5cc52661cf300f36b0a/pkg/encoding/v1/encoding_test.go#L147